### PR TITLE
ZOSVPC-3055- Give the option of select the image for a z/OS VSI.

### PR DESCRIPTION
**Pull request by Pketaki**
Updated files to customize the image for z/OS VSI for both Quickstart & Standard Variations.

Also, worked on indentation of JSON & .tf files.

### CODE REVIEW

File: solutions/standard/override.json [hunk 1]
```diff
@@ -845,33 +846,33 @@
       },
       {
          "boot_volume_encryption_key_name": null,
-         "image_name": "ibm-zos-2-5-s390x-dev-test-wazi-6",
+         "image_name": "ibm-zos-2-4-s390x-dev-test-wazi-10",
```
Comment (kevinkuhner) [inline solutions/standard/override.json:849]: Why did we change the default image_name value to a really old 2.4 image ?

File: solutions/standard/variables.tf [hunk 2]
```diff
@@ -44,6 +44,15 @@ variable "machine_type" {
   }
 }
 
+variable "image_name" {
+   description = "Enter a valid image name for Wazi VSI"
```
Comment (kevinkuhner) [inline solutions/standard/variables.tf:48]: See my comments in quickstart variation for the same variable.

**Review by kevinkuhner — CHANGES_REQUESTED** [review]
@Pketaki  - please see my comments.

File: solutions/quickstart/main.tf [hunk 3]
```diff
@@ -1,4 +1,12 @@
 ##############################################################################
+# Landing Zone Locals
+##############################################################################
+
+locals {
+  out = replace(var.override_json_string,"mz2o-2x16",var.machine_type)
+  image = replace(local.out,"ibm-zos-2-4-s390x-dev-test-wazi-10", var.image_name)
```
[the remark this answers is not in the document]
Comment (Pketaki) [reply solutions/quickstart/main.tf]: Updated the image_name variable. Added default value to image.

File: solutions/quickstart/variables.tf [hunk 4]
```diff
@@ -45,6 +45,15 @@ variable "machine_type" {
   }
 }
 
+variable "image_name" {
```
[the remark this answers is not in the document]
Comment (Pketaki) [reply solutions/quickstart/variables.tf:48]: Added default value.

File: solutions/quickstart/variables.tf [hunk 5]
```diff
@@ -45,6 +45,15 @@ variable "machine_type" {
...
+   description = "Enter a valid image name for Wazi VSI"
+   type        = string
+   validation {
+    condition     = can(regex("ibm-zos-[a-z]|[a-z][-a-z0-9]*[a-z0-9]", var.image_name)) || can(regex("zos-vsi-image-devtest-[-0-9]*[0-9]", var.image_name))
```
[the remark this answers is not in the document]
Comment (Pketaki) [reply solutions/quickstart/variables.tf]: Removed validation part and added default value.

File: solutions/quickstart/variables.tf [hunk 6]
```diff
@@ -264,7 +273,7 @@ variable "override_json_string" {
    "vsi": [
       {
          "boot_volume_encryption_key_name": null,
-         "image_name": "ibm-zos-2-5-s390x-dev-test-wazi-6",
+         "image_name": "ibm-zos-2-4-s390x-dev-test-wazi-10",
```
[the remark this answers is not in the document]
Comment (Pketaki) [reply solutions/quickstart/variables.tf]: Updated .

File: solutions/standard/main.tf [hunk 7]
```diff
@@ -24,14 +25,13 @@ locals {
...
-  ssh_public_key = var.ssh_public_key
-  override       = var.override
-  override_json_string                = local.out
+  source                 = "terraform-ibm-modules/landing-zone/ibm//patterns//vsi//module"
+  version                = "5.18.0"
+  prefix                 = var.prefix
+  region                 = var.region
+  ssh_public_key         = var.ssh_public_key
+  override               = var.override
+  override_json_string   = local.image
```
[the remark this answers is not in the document]
Comment (Pketaki) [reply solutions/standard/main.tf:34]: If customer gives custom image /stock image then this code will update the default with new input value

File: solutions/standard/override.json (hunk 1 above)
Comment (Pketaki) [reply solutions/standard/override.json]: Updated

File: solutions/standard/variables.tf (hunk 2 above)
Comment (Pketaki) [reply solutions/standard/variables.tf:48]: Added default value of image.

File: solutions/standard/main.tf [hunk 8]
```diff
@@ -11,11 +11,12 @@ provider "ibm" {
...
 
 ########################################################################################################################
-# Locals
+# Landing Zone Locals
 ########################################################################################################################
 
 locals {
-  out = replace(file("./override.json"), "mz2o-2x16", var.machine_type) 
+  out = replace(file("./override.json"), "mz2o-2x16", var.machine_type)
+  image = replace(local.out,"ibm-zos-2-4-s390x-dev-test-wazi-10", var.image_name)
```
[the remark this answers is not in the document]
Comment (Pketaki) [reply solutions/standard/main.tf]: If customer gives custom image /stock image then this code will update the default with new input value

File: solutions/quickstart/main.tf [hunk 9]
```diff
@@ -9,21 +17,22 @@ module "landing_zone" {
   prefix               = var.prefix
   region               = var.region
   ssh_key              = var.ssh_key
-  override_json_string = local.out
-}
-locals {
-  out = replace(var.override_json_string,"mz2o-2x16",var.machine_type)
+  override_json_string = local.image
```
[the remark this answers is not in the document]
Comment (Pketaki) [reply solutions/quickstart/main.tf:20]: If customer gives custom image /stock image then this code will update the default with new input value

**Review by kevinkuhner — APPROVED** [review]
@Pketaki  - I'm fine with this changes now.

**Comment by Pketaki** [comment]
closing the pR

**Pull request merged**